plex_generate_previews.py

```diff
@@ -74,10 +75,11 @@
 def generate_images(video_file, output_folder, lock):
     vf_parameters = "fps=fps={}:round=up,scale=w=320:h=240:force_original_aspect_ratio=decrease".format(round(1 / PLEX_BIF_FRAME_INTERVAL, 6))
     if MediaInfo.parse(video_file).video_tracks[0].hdr_format != "None":
-         vf_parameters = "fps=fps={}:round=up,zscale=t=linear:npl=100,format=gbrpf32le,zscale=p=bt709,tonemap=tonemap=hable:desat=0,zscale=t=bt709:m=bt709:r=tv,format=yuv420p,scale=w=320:h=240:force_original_aspect_ratio=decrease".format(round(1 / PLEX_BIF_FRAME_INTERVAL, 6))
+         if MediaInfo.parse(video_file).video_tracks[0].hdr_format is not None:
```

Comment:
This looks a little odd to me.

Are you trying to test id hdr_format exists? you should be able to do something like the below.. it will ensure media_info.video_tracks has elements in the array and ensure hdr_format is not None?

```
media_info = MediaInfo.parse(video_file)
if media_info.video_tracks and media_info.video_tracks[0].hdr_format

```

Reply:
Cleaned it up... I hope.

Reply:
Hmm, still not right.. Did you see my code snippet? 

```
    media_info = MediaInfo.parse(video_file)
    if media_info.video_tracks and media_info.video_tracks[0].hdr_format:
```

Reply:
is media_info.video_tracks[0].hdr_format a string None or a python None type?

Reply:
I saw your snippet. Your code has the advantage of a single call to mediainfo. I could change to a variable for a single call with two subsequent tests, but two lines of code.

The actual tests are different. Your snippet checks if there is video track information and if there is an HDR field. Any field, even a value of “None”, would satisfy that code.

A “properly” encoded file from most modern h264 or h265 encoders will have a field for HDR. If the field doesn’t exist at all, the file is SDR. If the field exists, the value is either “None” or some type of HDR encoding.  I originally checked only that the field was “None”. The problem encountered after cycling through many files is that if the field is non-existent, the original  conditional test will be satisfied and SDR files are tonemapped. That often results in a color space error. The new multi-conditional checks that either the field doesn’t exist at all or if it exists that it isn’t a value of None. In order to be HDR, we need the field present in some value other than “None”.

Reply:
That's right. A single call prevents it from having the read the media file from the disk twice. 

In python, None usually means it is null (https://realpython.com/null-in-python/) and will not pass an if statement test. So my check shouldn't let it continue if the field is set to None.

However, i noticed you are using "None" with quotes around it which means it is a string with a value of None. This is unusual and i would have expected it to set the value to None not a str('None').

Is it setting it to a string "None"

Reply:
That's right. A single call prevents it from having the read the media file from the disk twice. 

In python, None usually means it is null and will not pass an if statement test. So my check shouldn't let it continue if the field is set to None.

However, i noticed you are using "None" with quotes around it which means it is a string with a value of None. This is unusual and i would have expected it to set the value to None not a str('None').

Is it setting it to a string "None"

Reply:
Exactly. If the field doesn’t exist at all, it’s None. However, when the field exists, it is a string with the value of “None”.  Hence it’s either null and thus None or it’s “None” string value.  I will refactor for a single read.